### PR TITLE
Get subscribe number for PublisherProbe

### PR DESCRIPTION
**Pull request by mlevvy**
I would like to check how many somebody subscribed to my publisher.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@mlevvy Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/reactor?repositoryId=reactor/reactor-core&pullRequestId=1447)!

[Click here](https://cla.pivotal.io/sync/reactor?repositoryId=reactor/reactor-core&pullRequestId=1447) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

File: reactor-test/src/main/java/reactor/test/publisher/PublisherProbe.java [hunk 1]
```diff
@@ -131,6 +131,12 @@ default void assertWasRequested() {
 	 * @return true if the probe was subscribed to at least once.
 	 */
 	boolean wasSubscribed();
+
+	/**
+	 * @return how many times probe was subscribed
+	 */
+	long subscribed();
```
Comment (simonbasle) [inline reactor-test/src/main/java/reactor/test/publisher/PublisherProbe.java]: I would make this `long subscribeCount()`

File: reactor-test/src/main/java/reactor/test/publisher/ColdTestPublisher.java [hunk 2]
```diff
@@ -32,6 +33,7 @@
 import reactor.core.publisher.Mono;
 import reactor.core.publisher.Operators;
 import reactor.util.annotation.Nullable;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
```
[the remark this answers is not in the document]
Comment (mlevvy) [reply reactor-test/src/main/java/reactor/test/publisher/ColdTestPublisher.java]: OK

File: reactor-test/src/main/java/reactor/test/publisher/ColdTestPublisher.java [hunk 3]
```diff
@@ -232,7 +234,12 @@ void onComplete() {
 
 	@Override
 	public boolean wasSubscribed() {
-		return wasSubscribed;
+		return subscribed.get() > 0;
+	}
+
+	@Override
+	public long subscribed() {
+		throw new NotImplementedException();
```
[the remark this answers is not in the document]
Comment (mlevvy) [reply reactor-test/src/main/java/reactor/test/publisher/ColdTestPublisher.java]: My bad. Didn't write any test and forgot about it. Added some test and implementation.

File: reactor-test/src/main/java/reactor/test/publisher/PublisherProbe.java (hunk 1 above)
Comment (mlevvy) [reply reactor-test/src/main/java/reactor/test/publisher/PublisherProbe.java]: Ok

File: reactor-test/src/main/java/reactor/test/publisher/DefaultTestPublisher.java [hunk 4]
```diff
@@ -54,7 +55,7 @@
 
 	volatile boolean hasOverflown;
 	volatile boolean wasRequested;
-	volatile boolean wasSubscribed;
+	AtomicLong subscribed = new AtomicLong();
```
[the remark this answers is not in the document]
Comment (mlevvy) [reply reactor-test/src/main/java/reactor/test/publisher/DefaultTestPublisher.java]: Ok.

**Pull request merged**